language/move-lang/src/shared/mod.rs

```diff
@@ -227,3 +269,84 @@ pub fn format_delim<T: fmt::Display, I: IntoIterator<Item = T>>(items: I, delim:
 pub fn format_comma<T: fmt::Display, I: IntoIterator<Item = T>>(items: I) -> String {
     format_delim(items, ", ")
 }
+
+//**************************************************************************************************
+// Flags
+//**************************************************************************************************
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum Flag {
+    Test,
+}
+
+impl Flag {
+    pub const FLAGS: &'static [Flag] = &[Flag::Test];
```

Comment:
Should this be pub?

Reply:
I was making it pub so that there would be some way of reflecting over the set of flags that are accepted by compiler. Let me know if you still think it should be private and I can do that.